### PR TITLE
e2e: reduce dependency on /tmp for e2e tests

### PR DESCRIPTION
**Pull request by sstosh**
- Use filepath.Join(podmanTest.TempDir, "any") instead of "/tmp/any"

- Add generatePolicyFile() to avoid the hardcording of "keyPath": "tmp/key.gpg"

Signed-off-by: Toshiki Sonoda <sonoda.toshiki@fujitsu.com>

---

Related to #17089 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/e2e/save_test.go [hunk 1]
```diff
@@ -187,6 +187,7 @@ var _ = Describe("Podman save", func() {
 
 		cmd = exec.Command("cp", "sign/key.gpg", "/tmp/key.gpg")
 		Expect(cmd.Run()).To(Succeed())
+		defer os.Remove("/tmp/key.gpg")
```
Comment (Luap99) [inline test/e2e/save_test.go:190]: Why does this still hardcode `/tmp`?

**Review by Luap99 — COMMENTED** [review]
every test already has access to `podmanTest.TempDir`, I think we should stick to that instead.

**Comment by rhatdan** [comment]
I agree.

**Comment by edsantiago** [comment]
Very nice. Thank you for breaking this down into smaller parts! LGTM.

**Pull request merged**